src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
+
+When apps call [`redirectToSignIn`](http://blockstack.github.io/blockstack.js/index.html#redirecttosignin) or [`redirectToSignInWithAuthRequest`](http://blockstack.github.io/blockstack.js/index.html#redirecttosigninwithauthrequest), the code tries to check if a handler for the `blockstack:` custom protocol is installed and, if so, redirects the user to `blockstack:<authRequestToken>`. This causes the authentication request token to be passed from the app to the Blockstack Browser which in turn validates the request, displays information about the app to the user and asks the user if she wants to approve sign in to the app.
 
-### Verify an auth response
+## Adding Blockstack Authentication to your app
+
+The way you can add Blockstack Authentication to you app depends on whether your app is a modern decentralized Blockstack App where code runs client-side without trusted servers or a legacy client-server app where a server is trusted. 
+
+### Authentication in Client-side apps
+This method is appropriate for decentralized client-side apps where the user's zone of trust - the parts of the app that the user is trusting - begins and ends with the code running on their own computer. In apps like these, any code the app interacts with that's not on their own computer such as external servers does not need to know who she is.
+
+[Blockstack.js](https://github.com/blockstack/blockstack.js) provides API methods that help you to implement Blockstack Authentication in your client-side app.  
+
+#### Standard flow
+The preferred way to implement authentication in these apps is to use the standard flow. This flow hides much of the process behind a few easy function calls and makes it very fast to get up and running.
+
+In this process you'll use these four functions:
+
+- `redirectToSignIn`
+- `isSignInPending`
+- `handlePendingSignIn`
+- `loadUserData`
+
+##### Starting the sign in process
+
+When your app wants to start the sign in process, typically when the user clicks a "Sign in with Blockstack" button, your app will call the [`redirectToSignIn`](http://blockstack.github.io/blockstack.js/index.html#redirecttosignin) method of [blockstack.js](https://github.com/blockstack/blockstack.js). 
+
+This creates an ephemeral transit key, stores it in the web browser's `localStorage`, uses it to create an authentication request token and finally redirects the user to the Blockstack browser to approve the sign in request.
+
+##### Handling an authentication response
+
+When a user approves a sign in request, the Blockstack Browser will return the signed authentication response token to the `redirectURI` specified in `redirectToSignIn`. To check for the presence of this token, your app calls `isSignInPending` and if this returns `true` you then call the `handlePendingSignIn` which decodes the token and returns the signed in user's data and simultaneously storing it to `localStorage` so that it can be retrieved later with `loadUserData`.
```

Comment:
->

When a user approves a sign in request, the Blockstack Browser will return the signed authentication response token to the `redirectURI` specified in `redirectToSignIn`.

To check for the presence of this token, your app should call `isSignInPending`. If this returns `true`, the app should then call `handlePendingSignIn`. This decodes the token, returns the signed in user's data, and simultaneously stores it to `localStorage` so that it can be retrieved later with `loadUserData`.